pkgs/tools/inputmethods/ibus-engines/ibus-rime/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchFromGitHub, pkgconfig, cmake, librime, ibus, libnotify, brise }:
+
+stdenv.mkDerivation rec {
+  pname = "ibus-rime";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "rime";
+    repo = "ibus-rime";
+    rev = version;
+    sha256 = "0zbajz7i18vrqwdyclzywvsjg6qzaih64jhi3pkxp7mbw8jc5vhy";
+  };
+
+  # FIXME: Replace brise (deprecated) with rime/plum
+  buildInputs = [ pkgconfig cmake librime ibus libnotify brise ];
+
+  # cmake cannont automatically find our nonstandard brise install location
+  cmakeFlags = [ "-DRIME_DATA_DIR=${brise}/share/rime-data" ];
+
+  # Copied from Makefile as cmake will override it
+  installPhase = ''
```

Comment:
Whhy not use the CMake's default install phase?

Reply:
The repo (ibus-rime) provides install target in the hand-written Makefile instead of defining it in CMake.

So 've got to copy that script out

Maybe I'll make a PR to it later

Reply:
Then the comment should read CMake build scripts do not provide install target, or something.
Also it should be possible to use something like:

```
installFlags = [
  "-f" "../Makefile"
  "PREFIX=${placeholder ''out''}"
  "builddir=." # maybe not necessary not sure if make -f works relative to $PWD or the Makefile
];
```

Reply:
I've changed the comment. But can't figure out how to use `installFlags`
I'll just use the `installPhase`

Reply:
Something like this might work better:

```patch
--- a/pkgs/tools/inputmethods/ibus-engines/ibus-rime/default.nix
+++ b/pkgs/tools/inputmethods/ibus-engines/ibus-rime/default.nix
@@ -41,6 +41,10 @@ stdenv.mkDerivation rec {
     "-DRIME_DATA_DIR=${brise}/share/rime-data"
   ];
 
+  installFlags = [
+    "PREFIX=${placeholder ''out''}"
+  ];
+
   # Fix hard-coded paths references on rime-data and icons
   preBuild = ''
     sed -i ../rime_config.h -E \
@@ -48,17 +52,10 @@ stdenv.mkDerivation rec {
       -e "s|(IBUS_RIME_ICONS_DIR).+$|\1 \"$out/share/ibus-rime/icons\"|"
   '';
 
-  # CMake script does not provide install phrase
-  installPhase = ''
-    install -m 755 -d $out/share/ibus/component
-    install -m 644 -t $out/share/ibus/component/ ${src}/rime.xml
-
-    install -m 755 -d $out/lib/ibus-rime
-    install -m 755 -t $out/lib/ibus-rime/ ibus-engine-rime
-
-    install -m 755 -d $out/share/ibus-rime
-    install -m 755 -d $out/share/ibus-rime/icons
-    install -m 644 -t $out/share/ibus-rime/icons/ ${src}/icons/*.png
+  preInstall = ''
+    # CMake script does not provide install phrase,
+    # we need to use the Makefile in the source tree.
+    cd ..
   '';
 
   # The ibus config rime.xml hard-coded binary path for ibus-engine-rime
```

Of course, it would be best if the `install` target were added to `CMakeLists.txt`. I opened an issue about that: https://github.com/rime/ibus-rime/issues/108